modules/javafx.controls/src/main/java/javafx/scene/control/Skinnable.java

```diff
@@ -36,23 +36,30 @@
  */
 public interface Skinnable {
     /**
-     * Skin is responsible for rendering this {@code Control}. From the
-     * perspective of the {@code Control}, the {@code Skin} is a black box.
-     * It listens and responds to changes in state in a {@code Control}.
+     * Skin is responsible for rendering this {@code Skinnable}. From the
+     * perspective of the {@code Skinnable}, the {@code Skin} is a black box.
+     * It listens and responds to changes in state in a {@code Skinnable}.
      * <p>
-     * There is a one-to-one relationship between a {@code Control} and its
+     * There is typically a one-to-one relationship between a {@code Skinnable} and its
      * {@code Skin}. Every {@code Skin} maintains a back reference to the
-     * {@code Control}.
+     * {@code Skinnable}.
      * <p>
      * A skin may be null.
      *
-     * @return the skin property for this control
+     * @return the skin property for this Skinnable
      */
     public ObjectProperty<Skin<?>> skinProperty();
 
     /**
-     * Sets the skin that will render this {@link Control}
-     * @param value the skin value for this control
+     * Sets the skin that will render this {@link Skinnable}.
+     * <p>
+     * To ensure a one-to-one relationship between a {@code Skinnable} and its
+     * {@code Skin}, this method may check the return value of
+     * {@link Skin#getSkinnable()} against this Skinnable,
+     * and may throw an {@code IllegalArgumentException} if it is not the same.
+     *
+     * @param value the skin value for this Skinnable
+     * @throws IllegalArgumentException if {@code Skin} does not correspond to this {@code Skinnable}
      */
     public void setSkin(Skin<?> value);
```

Comment:
Generally docs for properties should go on just the property and not the setter or getter. The docs will be copied and cross linked by the javadoc tool. Can you try that, and generate a javadoc, and see if that holds in this case (we don't have many cases where a property is in an interface).

Reply:
So javadoc tool ignores the interface, resulting in Control.setSkin(skin) method inheriting the property's description.  

Curiously, eclipse does show the interface's version, which I think helps more than having three identical descriptions for the property, its getter and setter.

What would you recommend we do here?

Reply:
done